src/reducers/authReducer.ts

```diff
@@ -0,0 +1,32 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+interface AuthState {
+  user: {
+    token: string | null;
+  };
+}
+
+const initialState: AuthState = {
+  user: {
+    token: null,
+  },
+};
+
+const authSlice = createSlice({
+  name: 'counter',
+  initialState,
+  reducers: {
+
+    login: (state, actions) => ({
+      user: actions.payload.user,
+    }),
```

Comment:
```suggestion
    login: (state, action: PayloadAction<User>) => {
    state.user = action.payload;
}
```
This will make the code easier to read and write, and it ensures that the Redux store is updated efficiently and correctly. More context on this [Here](https://redux.js.org/usage/usage-with-typescript)

Reply:
Assignment to property of function parameter 'state'.eslint[no-param-reassign](https://eslint.org/docs/rules/no-param-reassign)

Reply:
Assignment to property of function parameter 'state'.eslint[no-param-reassign](https://eslint.org/docs/rules/no-param-reassign)